pkg/client/client.go

```diff
@@ -1526,42 +1527,27 @@ func (c *Client) CreateOrUpdateClusterRoleBinding(ctx context.Context, crb *rbac
 
 func (c *Client) CreateOrUpdateServiceAccount(ctx context.Context, sa *v1.ServiceAccount) error {
 	sClient := c.kclient.CoreV1().ServiceAccounts(sa.GetNamespace())
-	_, err := sClient.Get(ctx, sa.GetName(), metav1.GetOptions{})
+	existing, err := sClient.Get(ctx, sa.GetName(), metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
 		_, err := sClient.Create(ctx, sa, metav1.CreateOptions{})
 		return errors.Wrap(err, "creating ServiceAccount object failed")
 	}
-	return errors.Wrap(err, "retrieving ServiceAccount object failed")
-	// TODO(JoaoBraveCoding) The code below was reverted due to
-	// https://bugzilla.redhat.com/show_bug.cgi?id=2115527 see also
-	// https://bugzilla.redhat.com/show_bug.cgi?id=2095719 for more context
-	// if err != nil {
-	// 	return errors.Wrap(err, "retrieving ServiceAccount object failed")
-	// }
-
-	// required := sa.DeepCopy()
-	// mergeMetadata(&required.ObjectMeta, existing.ObjectMeta)
-
-	// // Why we use Patch here? ServiceAccounts get a new secret generated whenever
-	// // they are updated, even if nothing has changed. This is likely due to "Update"
-	// // performing a PUT call signifying, that this may be a new ServiceAccount,
-	// // therefore a new token is needed.
-	// oldData, err := json.Marshal(existing)
-	// if err != nil {
-	// 	return errors.Wrap(err, "marshaling existing ServiceAccount object failed")
-	// }
-
-	// newData, err := json.Marshal(required)
-	// if err != nil {
-	// 	return errors.Wrap(err, "marshaling updated ServiceAccount object failed")
-	// }
-
-	// patchBytes, patchErr := strategicpatch.CreateTwoWayMergePatch(oldData, newData, required)
-	// if patchErr != nil {
-	// 	return errors.Wrap(patchErr, "creating ServiceAccount two way merge patch failed")
-	// }
-	// _, err = sClient.Patch(ctx, required.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
-	// return errors.Wrap(err, "patching ServiceAccount object failed")
+	if err != nil {
+		return errors.Wrap(err, "retrieving ServiceAccount object failed")
+	}
+	modified := resourcemerge.BoolPtr(false)
```

Comment:
Nitpick: wouldn't it be simpler to just use the following?:

```go
modified := false
...
resourcemerge.EnsureObjectMeta(&modified, &existing.ObjectMeta, sa.ObjectMeta)
...
	if !modified {
```

Reply:
This could also work, I took inspiration from a go library that deals with the same problem https://github.com/openshift/library-go/blob/84c00ba9649a8c718023a43ace173e972656afee/pkg/operator/resource/resourceapply/core.go#L227

Same for the comment bellow 😅

Reply:
I think Juans suggestion would improve readbility.

Reply:
I wonder if we should use `ApplyServiceAccountImproved()` directly as duplicating the same logic seems error-prone?

Reply:
> I think Juans suggestion would improve readbility.

I'm okay with it, I just didn't do it to try to be as close as possible to the original code. 

> I wonder if we should use ApplyServiceAccountImproved() directly as duplicating the same logic seems error-prone?


Yes, I wanted to use it but the function requires a `recorder events.Recorder` which if I'm not mistaken we don't have in CMO